### PR TITLE
test: 💍 Add tests for core

### LINKED ISSUE #4: Implement tests
The project needs tests

- [x] core
- [ ] Docker
- [ ] K8S
- [ ] Git

### PR DESCRIPTION
**Pull request by MaethorNaur**
✅ Closes #4

### CODE REVIEW

File: core/src/test/resources/application.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+test = "configuration"
```
Comment (andrewrigas) [inline core/src/test/resources/application.conf:1]:
```suggestion
test = "valid"
```

File: core/src/test/scala/restui/ConfigurationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package restui
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.scalatest.compatible.Assertion
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class ConfigurationSpec extends AnyWordSpec with Matchers {
+  private def checkConfiguration(config: Config, value: String = "configuration"): Assertion =
```
Comment (andrewrigas) [inline core/src/test/scala/restui/ConfigurationSpec.scala:9]:
```suggestion
  private def checkConfiguration(config: Config, value: String = "valid"): Assertion =
```

File: core/src/test/scala/restui/ConfigurationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      "fallback to default configuration" when {
+        "the file does not exist" in {
+          checkConfiguration(Configuration.config(Some("not-existing-file.conf")))
+        }
+        "the file is not a valid HOCON file" in {
+          checkConfiguration(Configuration.config(Some("src/test/resources/invalid.conf")))
+        }
+      }
+      "get the value from the file" in {
+        checkConfiguration(Configuration.config(Some("src/test/resources/valid.conf")), "valid")
```
Comment (andrewrigas) [inline core/src/test/scala/restui/ConfigurationSpec.scala:26]:
```suggestion
        checkConfiguration(Configuration.config(Some("src/test/resources/valid.conf")))
```

File: core/src/test/scala/restui/models/ContentTypeSpec.scala [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+
+class ContentTypeSpec extends AnyFlatSpec with Matchers with TableDrivenPropertyChecks {
+  private val properties = Table(
+    ("file", "type"),
+    ("test.yaml", ContentType.Yaml),
+    ("test.yml", ContentType.Yaml),
+    ("test.json", ContentType.Json),
+    ("test.txt", ContentType.Plain)
```
Comment (andrewrigas) [inline core/src/test/scala/restui/models/ContentTypeSpec.scala:13]: edge cases ? and what about if reads something else ?

File: core/src/test/scala/restui/models/EventSpec.scala [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import restui.models.Event._
+
+class EventSpec extends AnyFlatSpec with Matchers with TableDrivenPropertyChecks {
+  private val properties = Table(
+    ("event", "json"),
+    (Event.ServiceUp("id", "test", Map("key" -> "value")), """{"event":"serviceUp","id":"id","name":"test","metadata":{"key":"value"}}"""),
+    (Event.ServiceDown("id"), """{"event":"serviceDown","id":"id"}"""),
+    (List(Event.ServiceUp("id", "test", Map("key" -> "value"))),
```
Comment (andrewrigas) [inline core/src/test/scala/restui/models/EventSpec.scala:14]: I would say list doesnt have to belog in this set  and maybe create another table with list of service up and service down

File: core/src/test/scala/restui/models/ContentTypeSpec.scala (hunk 4 above)
Comment (MaethorNaur) [reply core/src/test/scala/restui/models/ContentTypeSpec.scala:13]: There are 3 types:

- Yaml: `yaml`,`yml`
- Json: `json`
- Others: whatever
